packages/task/src/node/task-server.ts

```diff
@@ -6,134 +6,60 @@
-    protected tasksToDispose = new Map<number, Disposable>();
-
-    constructor(
-        @inject(ILogger) @named('task') protected readonly logger: ILogger,
-        @inject(RawProcessFactory) protected readonly rawProcessFactory: RawProcessFactory,
-        @inject(TerminalProcessFactory) protected readonly terminalProcessFactory: TerminalProcessFactory,
-        @inject(TaskManager) protected readonly taskManager: TaskManager,
-        @inject(TaskFactory) protected readonly taskFactory: TaskFactory
-    ) {
-        taskManager.onDelete(id => {
-            const toDispose = this.tasksToDispose.get(id);
-            if (toDispose !== undefined) {
-                toDispose.dispose();
-                this.tasksToDispose.delete(id);
-            }
-        });
-    }
+
+    @inject(ILogger) @named('task')
+    protected readonly logger: ILogger;
+
+    @inject(TaskManager)
+    protected readonly taskManager: TaskManager;
+
+    @inject(TaskRunnerRegistry)
+    protected readonly runnerRegistry: TaskRunnerRegistry;
 
     dispose() {
         // do nothing
     }
 
     getTasks(context?: string | undefined): Promise<TaskInfo[]> {
-        const taskinfo: TaskInfo[] = [];
+        const taskInfo: TaskInfo[] = [];
 
         const tasks = this.taskManager.getTasks(context);
         if (tasks !== undefined) {
             for (const task of tasks) {
-                taskinfo.push(task.getRuntimeInfo());
+                taskInfo.push(task.getRuntimeInfo());
             }
         }
-        this.logger.debug(`getTasks(): about to return task information for ${taskinfo.length} tasks`);
+        this.logger.debug(`getTasks(): about to return task information for ${taskInfo.length} tasks`);
 
-        return Promise.resolve(taskinfo);
+        return Promise.resolve(taskInfo);
     }
 
-    async run(options: TaskOptions, ctx?: string): Promise<TaskInfo> {
-        // on windows, prefer windows-specific options, if available
-        const processOptions = (isWindows && options.windowsProcessOptions !== undefined) ?
-            options.windowsProcessOptions : options.processOptions;
-
-        const command = processOptions.command;
-
-        // sanity checks:
-        // - we expect the cwd to be set by the client.
-        // - we expect processType to be set by the client
-        if (!options.cwd) {
-            return Promise.reject(new Error("Can't run a task when 'cwd' is not provided by the client"));
-        }
-        if (!options.processType) {
-            return Promise.reject(new Error("Can't run a task when 'processType' is not provided by the client"));
+    async run(taskConfiguration: TaskConfiguration, ctx?: string): Promise<TaskInfo> {
+        const taskType = taskConfiguration.type;
+        const runner = this.runnerRegistry.getRunner(taskType);
+        if (!runner) {
+            return Promise.reject(new Error(`No corresponding Runner found for the Task type ${taskType}`));
```

Comment:
It might be nice to have a unit test that covers this case here, where a task runner is not found

Reply:
Actually, this is a leftover from the previous version when there wasn't a default runner. I removed that code and added a test to check that a default runner is used.